app/views/pages/item-detail/index.js

```diff
@@ -0,0 +1,194 @@
+
+let initialItemDetailHeight = function() {
+  let bodyHeight = document.body.getBoundingClientRect().height;
+  let headerHeight = document.getElementsByClassName('product__header-menu')[0].getBoundingClientRect().height;
+
+  return bodyHeight - headerHeight;
+}
+
+var ItemDetail = React.createClass({
+
+  mixins: [State],
+
+  getInitialState() {
+    return {
+      item: {},
+      attachmentsPanel: false,
+      itemDetailHeight: initialItemDetailHeight(),
+      descriptionEditable: false,
+    };
+  },
+
+  updateStripeHeight() {
+    let content = document.getElementsByClassName('item-detail__content')[0];
+    let height = content ? content.getBoundingClientRect().height : 0;
+
+    if (height > this.state.itemDetailHeight) {
+      this.setState({itemDetailHeight: height})
+    }
+  },
+
+  componentWillReceiveProps: function(nextProps) {
+    this.updateStripeHeight();
+
+    if (this.getParams().number != this.state.item.number) {
+      ItemActions.fetchItem(this.getParams().id, this.props.number);
+      ItemActions.fetchActivity(this.getParams().id, this.getParams().number);
+      this._onChange();
+    }
+  },
+
+  setItem(key, ev, value) {
+    let item = _.cloneDeep(this.state.item);
+    item[key] = value;
+
+    this.setState({item: item});
+  },
+
+  updateItem() {
+    let productId = this.getParams().id;
+    let itemId = this.getParams().number;
+
+    ProductActions.updateItem(productId, itemId, { description: this.state.item.description });
+  },
+
+  itemDetails() {
+    return (
+      <ItemDetails  members={this.props.members}
+                       type={this.state.item.type}
+                      title={this.state.item.title}
+                        who={this.state.item.who}
+                       what={this.state.item.what}
+                        why={this.state.item.why}
+                     number={this.state.item.number}
+                       tags={this.state.item.tags}
+                  createdAt={this.state.item.created_at}
+                  createdBy={this.state.item.created_by}
+                     status={this.state.item.status}
+                     score={this.state.item.score}
+                   assignee={this.state.item.assigned_to} />
+    )
+  },
+
+  itemDescription() {
+    return (
+      <ItemDescription followers={[]}
+                     description={this.state.item.description}
+                         setItem={this.setItem} />
+    )
+  },
+
+  itemAttachments() {
+    let attachments = [];
+    let item = this.state.item;
+
+    if (item.activity && item.activity.activities) {
+      attachments = _.where(item.activity.activities, {'action':'attachment'});
+    }
+
+    return (
+      <ItemAttachments attachments={attachments} />
+    )
+  },
+
+  subitems() {
+    let subitems = this.state.item.sub_items || [];
+
+    return (
+      <ItemSubitems  members={this.props.members}
+                    subitems={subitems} />
+    )
+  },
+
+  itemComments() {
```

Comment:
So ItemComments should prob be named `CommentForm` :monkey_face: 

And it gets its state from the Router `State` mixin where its pulling the `product.id` and the `item.id`

It also manages its own internal comment state because its parent didn't really need to have it bubbled back up